### ISSUE TITLE
Assertions of  MultiBody.Examples.Loops.Utilities.GasForce2

### ISSUE THREAD

**Issue by modelica-trac-importer** [modelica/Modelica#1144]
**Reported by beutlich on 29 May 2013 08:40 UTC**
SimulationX throws the following assertion on Modelica.Mechanics.MultiBody.Examples.Loops.Engine1b
```
Error
In Base type Modelica.Mechanics.MultiBody.Examples.Loops.Utilities.Engine1bBase of gasForce: Assert failed:  flange_b.s - flange_a.s (= 0.35 <= L (0.35) required for GasForce component.
Most likely, parameter L is not correct.
```
I suggest to add format specifiers to the assertions of Modelica.Mechanics.MultiBody.Examples.Loops.Utilities.GasForce2 to reflect the difference of 1e-12 in the output message, e.g.
```
    assert(s_rel >= -1.e-12, "flange_b.s - flange_a.s (= " + String(s_rel, format=".16g") +
                             ") >= 0 required for GasForce component.\n" +
                             "Most likely, the component has to be flipped.");
    assert(s_rel <= L + 1.e-12, " flange_b.s - flange_a.s (= " + String(s_rel, format=".16g") +
                                " <= L (" + String(L, format=".16g") + ") required for GasForce component.\n" +
                                "Most likely, parameter L is not correct.");
```

----

Migrated-From: https://trac.modelica.org/Modelica/ticket/1144

**Comment by modelica-trac-importer** [comment]
**Comment by beutlich on 29 May 2013 08:49 UTC**
Or use the significant digits, e.g.
```
    assert(s_rel >= -1.e-12, "flange_b.s - flange_a.s (= " + String(s_rel, significantDigits=16) +
                             ") >= 0 required for GasForce component.\n" +
                             "Most likely, the component has to be flipped.");
    assert(s_rel <= L + 1.e-12, " flange_b.s - flange_a.s (= " + String(s_rel, significantDigits=16) +
                                " <= L (" + String(L, significantDigits=16) + ") required for GasForce component.\n" +
                                "Most likely, parameter L is not correct.");
```
which now gives
```
Error
In Base type Modelica.Mechanics.MultiBody.Examples.Loops.Utilities.Engine1bBase of gasForce: Assert failed:  flange_b.s - flange_a.s (= 0.3500000004331598 <= L (0.35) required for GasForce component.
Most likely, parameter L is not correct.
```

**Comment by modelica-trac-importer** [comment]
**Comment by beutlich on 29 May 2013 09:11 UTC**
There is also a missing closing bracket in the second assertion string.

**Comment by modelica-trac-importer** [comment]
**Comment by otter on 17 Jun 2013 10:29 UTC**
Fixed in 3da20b31f3a8ef8964221abb40aeaeb2bd2c1b4f and 6102b905b9074678879892c2cada85f470b61c7a as suggested
